Aquality.Selenium.Core/src/Aquality.Selenium.Core/Utilities/ActionRetrier.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Aquality.Selenium.Core.Configurations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Aquality.Selenium.Core.Utilities
+{
+    public class ActionRetrier : IActionRetrier
+    {
+        private readonly IRetryConfiguration retryConfiguration;
+
+        /// <summary>
+        /// Instantiates the class using retry configuration.
+        /// </summary>
+        /// <param name="retryConfiguration">Retry configuration.</param>
+        public ActionRetrier(IRetryConfiguration retryConfiguration)
+        {
+            this.retryConfiguration = retryConfiguration;
+        }
+
+        /// <summary>
+        /// Retries the action when one of the handledExceptions occures.
+        /// </summary>
+        /// <param name="action">Action to be applied.</param>
+        /// <param name="handledExceptions">Exceptions to be handled.</param>
+        public virtual void DoWithRetry(Action action, IEnumerable<Type> handledExceptions)
```

Comment:
won't the docs from the interface be automatically applied? do we still need to add docs in both places?

Reply:
Docs are not applied from interfaces to implementation in c#. However, for most cases we not duplicate docs from interfaces to implementation

Reply:
Actually, for most cases we duplicate docs from interfaces in dotnet libs